src/wfs/src/main/java/org/geoserver/wfs/request/Delete.java

```diff
@@ -102,4 +97,20 @@ protected void eAddForDelete(EObject obj, String property, Filter newFilter) {
             ((DeleteTypeImpl) obj).setFilter(result);
         }
     }
+
+    private List<Filter> combineFilters(Filter newFilter, Filter currentFilter) {
+        List<Filter> filters = new ArrayList<>();
+        if (newFilter != null) {
+            filters.add(newFilter);
+        }
+
+        if (currentFilter != null) {
+            if (currentFilter instanceof Or) {
+                filters.addAll(((Or) currentFilter).getChildren());
+            } else {
+                filters.add(currentFilter);
+            }
+        }
```

Comment:
I'd suggest to make this symmetrical, and apply the same logic to newFilter as well, just in case.
So the method could become a "addFilters(Filter, List<Filter> filters)" and called in sequence with both filter and currentFilter.